server/server.js

```diff
@@ -26,13 +26,16 @@ log("server started", { port: config.PORT });
 
 var CSP = "default-src 'self'; style-src 'self' 'unsafe-inline'; connect-src 'self' ws: wss:";
 
-var fileserver = serveStatic(config.WEBROOT, {
-	maxAge: 2 * 3600 * 1000,
-	setHeaders: function (res) {
-		res.setHeader("X-UA-Compatible", "IE=Edge");
-		res.setHeader("Content-Security-Policy", CSP);
-	}
-});
+var fileserver = function(request, response, errorHandleFunction) {
+	request.url = request.url.replace(config.URL_PREFIX_PATH, "");
```

Comment:
We should probably avoid to mutate the request object here

Reply:
And I'm not sure I understand why this would have to be changed... The front-end reverse proxy should already be rewriting paths, so you shouldn't be receiving anything different here than in the normal case. Or are you trying to get the server to serve the app directly on a different path, withouth a front-end reverse proxy ?

Reply:
This is to run it directly on a different path or rather behind a reverse proxy without rewriting the paths.

Where would you change the request url instead?

Reply:
What exactly is the interest of running it directly under a different path without a reverse proxy ? The interest of running it under a path is to be able to run other servers on other paths, isn't it ?

Reply:
Yes, the only difference it would make is not having to create rewrite rules in the proxy.
I initially assumed that we would need to specify the prefix anyway in the server configuration however I've come to think that this is not the case and it would work with the reverse proxy rewriting the urls as well.

Reply:
OK, so let's just simplify the code, and create a [wiki page](https://github.com/lovasoa/whitebophir/wiki) explaining how to set up the reverse proxy.

Reply:
I have attempted to set up nginx to rewrite the url. While that part was successful and I grabbed the prefix out of `window.location.pathname` to get the correct url for socket.io to connect to, I was unable to get a correct handshake.

Reply:
I have also just noticed that we still need the prefix path to be set to be able to deliver correct absolute URLs for the open graph metadata

Reply:
So, can you remove the special handling in the file server and socket.io server ?

Reply:
I have removed the need for setting the prefix and got it running under nginx, I will add my configuration and the example given by @lmazet to the wiki